### PR TITLE
refactor(payment): PI-650 moved Adyen googlepay payment strategy implementation outside core package

### PR DESCRIPTION
**Pull request by vitalii-koshovyi**
## What?
Moved Adyen googlepay payment strategy implementation outside core package

## Why?
Due to the google pay refactring

## Testing / Proof
Units/manually
Adyen v2:

https://github.com/bigcommerce/checkout-sdk-js/assets/79574476/4fa9f78a-3bf8-4d94-9c62-6f16d2271cff

https://github.com/bigcommerce/checkout-sdk-js/assets/79574476/1db1b3af-572f-4208-9d93-00be605575cf

Adyen v3:

https://github.com/bigcommerce/checkout-sdk-js/assets/79574476/f6f7a6df-57e4-435a-9c7c-25159be3fa8a

https://github.com/bigcommerce/checkout-sdk-js/assets/79574476/1206ae34-57ee-4e6f-8965-be82273b3c81

@bigcommerce/team-checkout @bigcommerce/team-payments

### CODE REVIEW

**Review by bc-peng — APPROVED** [review]
LGTM.👏👏

File: packages/google-pay-integration/src/gateways/google-pay-adyenv3-gateway.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    NotInitializedError,
+    NotInitializedErrorType,
+    Payment,
+    PaymentIntegrationSelectors,
+    PaymentIntegrationService,
+    PaymentMethod,
+    PaymentMethodCancelledError,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+// import { assertsIsGooglePayAdyenV3InitializationData } from '../guards/is-google-pay-adyen-initialization-data';
```
Comment (serhii-tkachenko) [inline packages/google-pay-integration/src/gateways/google-pay-adyenv3-gateway.ts:24]: Do you still need this comment?

File: packages/google-pay-integration/src/gateways/google-pay-adyenv2-gateway.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    NotInitializedError,
+    NotInitializedErrorType,
+    Payment,
+    PaymentIntegrationSelectors,
+    PaymentIntegrationService,
+    PaymentMethod,
+    PaymentMethodCancelledError,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+// import { assertsIsGooglePayAdyenV2InitializationData } from '../guards/is-google-pay-adyen-initialization-data';
```
Comment (serhii-tkachenko) [inline packages/google-pay-integration/src/gateways/google-pay-adyenv2-gateway.ts:24]: Do you still need this comment?

File: packages/google-pay-integration/src/gateways/google-pay-adyenv3-gateway.ts (hunk 1 above)
Comment (serhii-tkachenko) [reply packages/google-pay-integration/src/gateways/google-pay-adyenv3-gateway.ts:24]: If not, I think, its ok to remove `assertsIsGooglePayAdyenV3InitializationData` method from code base

File: packages/google-pay-integration/src/gateways/google-pay-adyenv2-gateway.ts (hunk 2 above)
Comment (serhii-tkachenko) [reply packages/google-pay-integration/src/gateways/google-pay-adyenv2-gateway.ts:24]: If not, I think, its ok to remove `assertsIsGooglePayAdyenV2InitializationData` method from code base

**Review by serhii-tkachenko — APPROVED** [review]

**Review by PavlenkoM — APPROVED** [review]

**Pull request merged**